apps/api/src/donations/donations.service.ts

```diff
@@ -167,4 +167,19 @@ export class DonationsService {
     }, 0)
     return { donations, total }
   }
+
+  async getRecentDonationsByCampaign(campaignId: string) {
+    const vaults = await this.prisma.vault.findMany({ where: { campaignId } })
+    const donations = await this.prisma.donation.findMany({
+      where: {
```

Comment:
Request to db for getting vaults could be skipped with joining vault table. Where clause could be modified 
where: { targetVault: { campaignId },  status: DonationStatus.succeeded }

Reply:
Thank you for the optimization.